README.md

```diff
@@ -1,138 +1,615 @@
+
+```mjs
+class SuperClass {
+  set x(value) { console.log(value); }
+}
+
+class SubClassA extends SuperClass {
+  x = 1;
+}
+
+class SubClassB extends SuperClass {
+  @set x = 1;
+}
+
+const a = new SubClassA();  // does not log anything
+a.x;                        // 1
+
+const b = new SubClassB();  // logs 1
+b.x;                        // undefined
+```
+
+The `@set` decorator is implemented with `@initialize`, which can decorate public fields. `@initialize` takes a callback as an argument, which is called after the field initializer is evaluated,
+
+```mjs
+// set.mjs
+
+export decorator @set { @initialize(function(value, key) { this[key] = value }) }
+```
+
+### `@tracked`
+
+The `@tracked` decorator turns a public field declaration into a getter/setter pair which triggers a `render()` method when the setter is called. This pattern, or patterns like it, is common in frameworksto avoid extra bookkeeping scattered throughout the application to ask for re-rendering.
+
+```mjs
+import { @tracked } from "./tracked.mjs";
+
+class Element {
+  @tracked counter = 0;
+
+  increment() { this.counter++; }
+
+  render() { console.log(counter); }
+}
+
+const e = new Element();
+e.increment();  // logs 1
+e.increment();  // logs 2
+```
+
+`@tracked` is defined in terms of the combination of two built-in decorators that we've seen before above. `@initialize` is used to replace the property definition with setting a property which stores the underlying data. `@register` is used to define a getter/setter pair which is used when accessing the property.
+
+```mjs
+// tracked.mjs
+
+export decorator @tracked {
+  @initialize(function(value, name) { this[`__internal_${name}`] = value; })
+  @register((target, name) => {
+    Object.defineProperty(target, "name", {
+      get() { return this[`__internal_${name}`]; },
+      set() { this[`__internal_${name}`] = value; this.render(); },
+      configurable: true
+    });
+  })
+}
+```
+
+Note, further built-in decorators as in [NEXTBUILTINS.md](./NEXTBUILTINS.md) may provide a more direct and statically analyzable way to implement `@tracked` and avoid the use of `Object.defineProperty`. This version unfortunately relies on metaprogramming when the class is defined.
+
+### `@bound`
+
+The `@bound` decorator makes a method auto-bound: it will carry around the original `this` value when accessed as `this.method` and not immediately called. This behavior matches Python's semantics, and it's been found useful in the React ecosystem, which makes frequent use of passing functions around. Example usage:
+
+```mjs
+import { @bound } from "./bound.mjs";
+
+class Foo {
+  x = 1;
+
+  @bound method() { console.log(this.x); }
+
+  queueMethod() { setTimeout(this.method, 1000); }
+}
+
+new Foo().queueMethod();  // will log 1, rather than undefined
+```
+
+One possible implementation, based on `@register`:
```

Comment:
is there a way to create a public field here instead of using a getter? `@bound` is entirely useless to me if it involves getters/setters in any way; i'd need it to put an *unbound* method on the prototype (which is also important for normal JS idioms of borrowable methods) and have *each instance* have a bound own property.

Reply:
More discussion of better techniques in NEXTBUILTINS.md

Reply:
To be clear; I consider `@bound` providing an own bound method an important use case for the initial iteration of decorators.

Reply:
I consider `@bound` important as well. I've pushed NEXTBUILTINS.md; I'd be curious what you think of the two approaches listed there.